bankaccount.go

```diff
@@ -37,6 +37,9 @@ type BankAccountParams struct {
 	// Token is a token referencing an external account like one returned from
 	// Stripe.js.
 	Token string `form:"-"`
+
+	// ID is used when tokenizing a bank account for shared customers
+	ID string `form:"*"`
```

Comment:
Do know if the `*` is required here? If we just need it for `AppendToAsSourceOrExternalAccount`, it might be worth changing to `-` to indicate that it's never needed elsewhere.

Reply:
@brahn-stripe with `-` the field seems ignored. Not sure if I'm making another mistake but it's not sent to stripe-mock in that case

Reply:
sorry that was for @brandur-stripe

Reply:
Ah okay, this was more of a question I guess. If you look at `BankAccountParams`, it's basically serialized in two separate ways:

1. When it's used for `bankaccount` or `token` operations, the default `form` implementation is used.
2. In certain cases (e.g. recipients), we call the special `AppendToAsSourceOrExternalAccount` function.

I just want to make sure that we're implementing the minimal logic that we need so that we don't make this anymore complicated. If we need the new `ID` field for just (1) above, then we should remove the changes in `AppendToAsSourceOrExternalAccount`. If we need it for just (2) (I don't think this is the case because you implemented a new test for `token`), then we should have just `form:"-"` to indicate that this field is never needed by `form`. If we need `ID` to be encoded for both (1) and (2), then we should leave this as is.

Can you just confirm that the latter is the case?

Reply:
@brandur-stripe My understanding was that we call `AppendToAsSourceOrExternalAccount` even for Token but I see now that this is wrong and I did not need to add the if inside the custom append. I will fix